recipes/fmilib/all/patches/2.4.1-003-use-expat-from-conan.patch

```diff
@@ -0,0 +1,110 @@
+-    -DEXPAT_SHARED_LIBS:BOOLEAN=OFF
+-    -DEXPAT_DTD:BOOLEAN=OFF
+-    -DEXPAT_NS:BOOLEAN=OFF
+-    -DEXPAT_MSVC_STATIC_CRT:BOOLEAN=${FMILIB_BUILD_WITH_STATIC_RTLIB}
+-    -DCMAKE_DEBUG_POSTFIX:STRING=
+-    -DCMAKE_RELEASE_POSTFIX:STRING=
+-    -DCMAKE_MINSIZEREL_POSTFIX:STRING=
+-    -DCMAKE_RELWITHDEBINFO_POSTFIX:STRING=
+-    -DCMAKE_POSITION_INDEPENDENT_CODE:BOOLEAN=ON
+-    -DCMAKE_BUILD_TYPE:STRING=${CMAKE_BUILD_TYPE}
+-    -DCMAKE_EXE_LINKER_FLAGS:STRING=${CMAKE_EXE_LINKER_FLAGS}
+-    -DCMAKE_LINK_LIBRARY_FLAG:STRING=${CMAKE_LINK_LIBRARY_FLAG}
+-    -DCMAKE_MODULE_LINKER_FLAGS:STRING=${CMAKE_MODULE_LINKER_FLAGS}
+-    -DCMAKE_SHARED_LINKER_FLAGS:STRING=${CMAKE_SHARED_LINKER_FLAGS}	
+-    -DCMAKE_INSTALL_PREFIX:PATH=${CMAKE_BINARY_DIR}/ExpatEx/install
+-)
+-
+-ExternalProject_Add(
+-	expatex
+-	PREFIX "${FMIXML_EXPAT_DIR}"
+-	SOURCE_DIR "${FMIXML_EXPAT_DIR}"
+-	CMAKE_CACHE_ARGS ${EXPAT_SETTINGS}
+-	BINARY_DIR ${CMAKE_BINARY_DIR}/ExpatEx
+-	INSTALL_DIR ${CMAKE_BINARY_DIR}/ExpatEx/install
+-	TMP_DIR     ${CMAKE_BINARY_DIR}/ExpatEx/tmp
+-    STAMP_DIR   ${CMAKE_BINARY_DIR}/ExpatEx/stamp
+-)
+-
+-ExternalProject_Add_Step(
+-	expatex dependent_reconfigure
+-	DEPENDEES configure
+-	DEPENDERS build
+-	COMMAND ${CMAKE_COMMAND} -E echo "Running:  ${CMAKE_COMMAND} -G "${CMAKE_GENERATOR}"  ${EXPAT_SETTINGS} ${FMIXML_EXPAT_DIR}"
+-	COMMAND ${CMAKE_COMMAND} -G "${CMAKE_GENERATOR}" ${EXPAT_SETTINGS} "${FMIXML_EXPAT_DIR}"
+-	DEPENDS ${CMAKE_BINARY_DIR}/CMakeCache.txt
+-	WORKING_DIRECTORY ${CMAKE_BINARY_DIR}/ExpatEx
+-)
+-
+-add_dependencies(expatex ${CMAKE_BINARY_DIR}/CMakeCache.txt ${FMILIBRARYHOME}/CMakeLists.txt)
+-
+-
+-if(MSVC)
+-    # Expat uses special naming with MSVC, which is mirrored here.
+-    set(EXPAT_LIB_PREFIX lib)
+-else()
+-    set(EXPAT_LIB_PREFIX ${CMAKE_STATIC_LIBRARY_PREFIX})
+-endif()
+-  
+-set(expatlib "${CMAKE_BINARY_DIR}/ExpatEx/${CMAKE_CFG_INTDIR}/${EXPAT_LIB_PREFIX}expat${CMAKE_STATIC_LIBRARY_SUFFIX}")
+-  
+-add_library(expat STATIC IMPORTED)
+-
+-set_target_properties(
+-	expat PROPERTIES 
+-		IMPORTED_LOCATION "${expatlib}"
+-)
+-
+-add_dependencies(expat expatex)
+-
+-if(FMILIB_INSTALL_SUBLIBS)
+-	install(FILES 
+-	"${CMAKE_BINARY_DIR}/ExpatEx/install/lib/${CMAKE_STATIC_LIBRARY_PREFIX}expat${CMAKE_STATIC_LIBRARY_SUFFIX}"
+-	DESTINATION lib)
+-endif()
+-
+-set(EXPAT_INCLUDE_DIRS ${CMAKE_BINARY_DIR}/ExpatEx/install/include)
+-
+-include_directories("${EXPAT_INCLUDE_DIRS}" "${FMILIB_THIRDPARTYLIBS}/FMI/" "${FMIXMLGENDIR}/FMI1" "${FMIXMLGENDIR}/FMI2")
++find_package(expat CONFIG REQUIRED)
++include_directories("${FMILIB_THIRDPARTYLIBS}/FMI/" "${FMIXMLGENDIR}/FMI1" "${FMIXMLGENDIR}/FMI2")
+ 
+ PREFIXLIST(FMIXMLSOURCE  ${FMIXMLDIR}/)
+ PREFIXLIST(FMIXMLHEADERS ${FMIXMLDIR}/)
+@@ -232,6 +157,6 @@ if(MSVC)
+     target_compile_definitions(fmixml PUBLIC XML_STATIC)
+ endif()
+ 
+-target_link_libraries(fmixml ${JMUTIL_LIBRARIES} expat)
++target_link_libraries(fmixml ${JMUTIL_LIBRARIES} expat::expat)
```

Comment:
This can probably be replaced with a `cd.set_property("expat", "cmake_target_name", "expat")`.

Reply:
Applied your suggesting.